omod/src/main/java/org/openmrs/module/spa/SpaDirectoryResolver.java

```diff
@@ -23,7 +26,19 @@ public class SpaDirectoryResolver implements GlobalPropertyListener {
 
     private static final AtomicReference<String> spaDirectory = new AtomicReference<>();
 
-    public SpaDirectoryResolver() {
+    private static ServletContext servletContext;
+
+    public SpaDirectoryResolver(ServletContext servletContext) {
+        this.servletContext = servletContext;
+        if (isBundledFrontend()){
```

Comment:
This kind of forces things to use the bundled frontend if one exists. I think a better solution here would be:

1. At module start, check to see if the local directory exists
2. If it does, there's no work to do
3. If it doesn't and there is a bundled frontend, extract it to the local directory

That way all the work is done in the Activator.